pkg/apiserver/registry/networkpolicy/clustergroupmember/rest.go

```diff
@@ -71,9 +77,59 @@ func (r *REST) Get(ctx context.Context, name string, options *metav1.GetOptions)
 		memberList.EffectiveMembers = effectiveMembers
 	}
 	memberList.Name = name
-	return memberList, nil
+	memberList.TotalMembers = int64(len(memberList.EffectiveMembers))
+	err = paginateMemberList(memberList, getOptions)
+	return memberList, err
+}
+
+func (r *REST) NewGetOptions() (runtime.Object, bool, string) {
+	return &controlplane.PaginationGetOptions{}, false, ""
 }
 
 func (r *REST) NamespaceScoped() bool {
 	return false
 }
+
+// paginateMemberList returns paginated results if meaningful options are provided.
+// Paginated results are continuous only when there is no member change across multiple calls.
+// Pagination is not processed if page number or limit = 0, thus returns full member list.
+// Returns an error for invalid options; returns empty list for page number beyond the total pages range.
+func paginateMemberList(memberList *controlplane.ClusterGroupMembers, pageInfo *controlplane.PaginationGetOptions) error {
+	if memberList.TotalMembers == 0 {
+		memberList.TotalPages, memberList.CurrentPage = 0, 0
+		return nil
+	}
+	if pageInfo.Limit < 0 {
+		return errors.NewBadRequest(fmt.Sprintf("received invalid page limit %d for pagination", pageInfo.Limit))
+	} else if pageInfo.Page < 0 {
+		return errors.NewBadRequest(fmt.Sprintf("received invalid page number %d for pagination", pageInfo.Page))
+	}
+	if pageInfo.Limit == 0 {
+		memberList.TotalPages, memberList.CurrentPage = 1, 1
+		return nil
+	}
+	totalPages := (int64(len(memberList.EffectiveMembers)) + pageInfo.Limit - 1) / pageInfo.Limit
+	memberList.TotalPages = totalPages
+	memberList.CurrentPage = pageInfo.Page
+	if totalPages >= pageInfo.Page && pageInfo.Page > 0 {
+		// Sort members based on name of ee/pod to realize consistent pagination support.
+		sort.SliceStable(memberList.EffectiveMembers, func(i, j int) bool {
+			if memberList.EffectiveMembers[i].Pod != nil && memberList.EffectiveMembers[j].Pod != nil {
+				return memberList.EffectiveMembers[i].Pod.Name < memberList.EffectiveMembers[j].Pod.Name
+			} else if memberList.EffectiveMembers[i].ExternalEntity != nil && memberList.EffectiveMembers[j].ExternalEntity != nil {
+				return memberList.EffectiveMembers[i].ExternalEntity.Name < memberList.EffectiveMembers[j].ExternalEntity.Name
+			} else {
+				return true
+			}
+		})
+		beginMember := (pageInfo.Page - 1) * pageInfo.Limit
+		memberList.EffectiveMembers = memberList.EffectiveMembers[beginMember:]
+		if pageInfo.Limit < int64(len(memberList.EffectiveMembers)) {
+			memberList.EffectiveMembers = memberList.EffectiveMembers[:pageInfo.Limit]
+		}
+	} else if totalPages < pageInfo.Page {
```

Comment:
if Page is 0, it seems it will get all members?

Reply:
Yes, it will be treated as no pagination option is provided, same as `limit = 0`.